src/CachePlugin.php

```diff
@@ -169,6 +180,34 @@ private function getMaxAge(ResponseInterface $response)
             return (new \DateTime($header))->getTimestamp() - (new \DateTime())->getTimestamp();
         }
 
-        return $this->defaultTtl;
+        return $this->config['default_ttl'];
+    }
+
+    /**
+     * Configure an options resolver.
+     *
+     * @param OptionsResolver $resolver
+     *
+     * @return array
+     */
+    private function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'default_ttl' => null,
+            'respect_cache_headers' => true,
+        ]);
+
+        $resolver->setAllowedTypes('default_ttl', ['int', 'null']);
+        $resolver->setAllowedTypes('respect_cache_headers', 'bool');
+    }
+
+    /**
+     * Set config to the plugin. This will overwrite any previously set config values.
+     *
+     * @param array $config
+     */
+    public function setConfig(array $config)
+    {
+        $this->config = $this->optionsResolver->resolve($config);
```

Comment:
i would inline that in the constructor. there is no need to change the configuration at runtime. and then we can make the OptionsResolver a local class.